src/cloudai/_core/action_space.py

```
@@ -0,0 +1,55 @@
+    """
+
+    model_config = ConfigDict(extra="forbid")
+
+    low: float
+    high: float
+    dtype: Literal["int", "float"] = "float"
+
+    @model_validator(mode="after")
+    def _validate_bounds(self) -> Self:
+        if self.low >= self.high:
+            raise ValueError(f"ContinuousSpace requires low < high; got low={self.low}, high={self.high}")
```

Ruff (0.15.17)

[warning] Avoid specifying long messages outside the exception class

(TRY003)